OrbitLinuxTracing/TracerThread.cpp

```diff
@@ -729,6 +732,26 @@ void TracerThread::ProcessDeferredEvents() {
   }
 }
 
+void TracerThread::UpdateThreadNamesIfDelayElapsed() {
+  if (last_thread_names_update +
+          THREAD_NAMES_UPDATE_DELAY_MS * NS_PER_MILLISECOND <
+      MonotonicTimestampNs()) {
+    for (pid_t tid : ListThreads(pid_)) {
+      std::string name = GetThreadName(tid);
+      if (name.empty()) {
+        continue;
+      }
+
+      auto last_name_it = thread_names_.find(tid);
+      if (last_name_it == thread_names_.end() || name != last_name_it->second) {
+        thread_names_[tid] = name;
+        listener_->OnThreadName(tid, name);
+      }
+    }
+    last_thread_names_update = MonotonicTimestampNs();
```

Comment:
MonotonicTimestampNs() could be called once at the top and the value be reused here.

Reply:
Done.